src/util/permissions.tsx

```diff
@@ -0,0 +1,311 @@
+import { LxdIdentity, LxdPermission } from "types/permissions";
+import { OptionHTMLAttributes } from "react";
+import { LxdImage } from "types/image";
+
+export const defaultOption = {
+  disabled: true,
+  label: "Select an option",
+  value: "",
+};
+
+export const noneAvailableOption = {
+  disabled: true,
+  label: "None available",
+  value: "",
+};
+
+// the resource types comes from the openFGA authorisation model in lxd
+// ref: https://discourse.ubuntu.com/t/identity-and-access-management-for-lxd/41516
+export const resourceTypeOptions = [
+  {
+    ...defaultOption,
+  },
+  {
+    value: "server",
+    label: "Server",
+  },
+  {
+    value: "identity",
+    label: "Identity",
+  },
+  {
+    value: "group",
+    label: "Group",
+  },
+  {
+    value: "certificate",
+    label: "Certificate",
+  },
+  {
+    value: "project",
+    label: "Project",
+  },
+  {
+    value: "profile",
+    label: "Profile",
+  },
+  {
+    value: "instance",
+    label: "Instance",
+  },
+  {
+    value: "image",
+    label: "Image",
+  },
+  {
+    value: "image_alias",
+    label: "Image alias",
+  },
+  {
+    value: "storage_pool",
+    label: "Storage pool",
+  },
+  {
+    value: "storage_volume",
+    label: "Storage volume",
+  },
+  {
+    value: "storage_bucket",
+    label: "Storage bucket",
+  },
+  {
+    value: "network",
+    label: "Network",
+  },
+  {
+    value: "network_acl",
+    label: "Network ACL",
+  },
+  {
+    value: "network_zone",
+    label: "Network zone",
+  },
+];
+
+const sortOptions = (
+  a: OptionHTMLAttributes<HTMLOptionElement>,
+  b: OptionHTMLAttributes<HTMLOptionElement>,
+) => {
+  if (b.label === "Select an option") {
+    return 1;
+  }
+
+  return (a.label ?? "").localeCompare(b.label as string);
+};
+
+export const generateResourceAndEntitlementOptions = (
+  resourceType: string,
+  permissions?: LxdPermission[],
+) => {
+  if (!permissions || !resourceType) {
+    return {
+      resourceEntitlementOptions: {},
+      resourceOptions: [],
+    };
+  }
+
+  const resourceEntitlementOptions: Record<
+    string,
+    OptionHTMLAttributes<HTMLOptionElement>[]
+  > = {};
+
+  for (const permission of permissions) {
+    if (permission.entity_type !== resourceType) {
+      continue;
+    }
```

Comment:
Do we need this? I think we fetch the permissions for the selected resourceType so this should never happen.
```suggestion
```

Reply:
Yeah you are right :+1: